catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DumpCommand.java

```diff
@@ -226,8 +233,6 @@ protected Object executeWithSubject() throws Exception {
     final AtomicLong resultCount = new AtomicLong(0);
```

Comment:
✏️ Using an atomic as a side effect to streaming should be a last resort. How about getting `handleResult()` to return its count and sum those up as part of the streaming?

Reply:
`handleResult()` may not always have that information. In the case that the dump command performs it's work multi-threaded, handleResult returns when the task is submitted to the executor, not when it's completed.

Reply:
Fair enough. However, because it can be done asynchronously and because we extract a value before calling `printStatus()` from the atomic object, it is entirely possible that `printStatus()` ends up printing in non sequential order.

Reply:
What line of code are you referencing? All of the count printouts are after this: https://github.com/codice/ddf/blob/39dd4c115af5dc61090abc5e53cc7c1212d736ab/catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DumpCommand.java#L262

Reply:
Because we are submitting multiple requests to an executor which uses a thread factory, one cannot guarantee that all interactions are done in sequence. So it is entirely possible that one thread collects a value from the AtomicLong to call `printStatus()` but another thread collects the next value after the first thread and ends up calling `printStatus()` and executing that code before. In such scenario, the values would be printed out of sync.

Reply:
What code do you see doing this? `printStatus` is only used in the Zip handling which is not called from the executor in the `if (StringUtils.isNotBlank(zipFileName)) {`. The executor is in `else if (multithreaded > 1) `. I don't see where printStatus is being called in a way that it would have the wrong value. Nothing in `processResults` call path calls `printStatus` nor does anything in that actually write the count.